src/libs/actions/Report.ts

```diff
@@ -2488,6 +2488,128 @@ function getReportPrivateNote(reportID: string) {
+    conciergeChatReportID = conciergeChatReport?.reportID;
+
+    const report = ReportUtils.getReport(conciergeChatReportID);
+
+    if (isNotEmptyObject(report) && ReportUtils.getReportNotificationPreference(report) === CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN) {
+        optimisticReport.notificationPreference = CONST.REPORT.NOTIFICATION_PREFERENCE.ALWAYS;
+    }
+
+    // Optimistically add the new actions to the store before waiting to save them to the server
+    const optimisticReportActions: OnyxCollection<NullishDeep<ReportAction>> = {};
+    if (reportCommentAction?.reportActionID) {
+        optimisticReportActions[reportCommentAction.reportActionID] = reportCommentAction;
+    }
+
+    type CompleteEngagementParameters = {
+        reportID: string;
+        reportActionID?: string;
+        commentReportActionID?: string | null;
+        reportComment?: string;
+        engagementChoice: string;
+        timezone?: string;
+    };
+
+    const parameters: CompleteEngagementParameters = {
+        reportID: conciergeChatReportID ?? '',
+        reportActionID: reportCommentAction.reportActionID,
+        reportComment: reportCommentText,
+        engagementChoice: choice,
+    };
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${conciergeChatReportID}`,
+            value: optimisticReport,
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${conciergeChatReportID}`,
+            value: optimisticReportActions,
+        },
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.NVP_INTRO_SELECTED,
+            value: {choice},
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${conciergeChatReportID}`,
+            value: {[reportCommentAction.reportActionID ?? '']: {pendingAction: null}},
+        },
+    ];
+
+    // Update the timezone if it's been 5 minutes from the last time the user added a comment
+    if (DateUtils.canUpdateTimezone()) {
+        const timezone = DateUtils.getCurrentTimezone();
+        parameters.timezone = JSON.stringify(timezone);
+        optimisticData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: ONYXKEYS.PERSONAL_DETAILS_LIST,
+            value: {[currentUserAccountID]: {timezone}},
+        });
+        DateUtils.setTimezoneUpdated();
+    }
+
+    API.write(commandName, parameters, {
+        optimisticData,
+        successData,
+    });
+    notifyNewAction(conciergeChatReportID ?? '', reportCommentAction.actorAccountID, reportCommentAction.reportActionID);
+}
+
+function dismissEngagementModal() {
+    const commandName = 'SetNameValuePair';
+    const parameters = {
+        name: ONYXKEYS.NVP_HAS_DISMISSED_IDLE_PANEL,
```

Comment:
Just curious but why is this an "idle panel" and not "engagement modal"?

Reply:
This was created for the OldDot version and is being re-used here because we want completing ether engagement modal to prevent both from showing. 

On the oldDot version it shows if you don't create a policy within 10s (if you just open the page and idle). That was the plan here too but the scope shifted